x-pack/legacy/plugins/ml/server/new_platform/licence_check_pre_routing_factory.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  KibanaRequest,
+  KibanaResponseFactory,
+  RequestHandler,
+  RequestHandlerContext,
+} from 'src/core/server';
+import { PLUGIN_ID } from './plugin';
+
+export const licensePreRoutingFactory = (
+  xpackMainPlugin: any,
+  handler: RequestHandler<any, any, any>
+): RequestHandler<any, any, any> => {
+  // License checking and enable/disable logic
+  return function licensePreRouting(
+    ctx: RequestHandlerContext,
+    request: KibanaRequest,
+    response: KibanaResponseFactory
+  ) {
+    const licenseCheckResults = xpackMainPlugin.info.feature(PLUGIN_ID).getLicenseCheckResults();
+
+    if (!licenseCheckResults.isAvailable) {
+      return response.customError({
+        body: {
+          message: licenseCheckResults.message,
+        },
+        statusCode: 403,
```

Comment:
You don't need to use `customError` for common return types. Use `response.forbidden` instead

Reply:
Makes sense - updated in 1be4d620155fc0186cd7fedcaebb6c7ba5a51c9c

Reply:
Updated to use `forbidden` in 8c93d6b4a20f1de2821b6ab6f8835703bdb09abf